whelktool/scripts/cleanups/2020/04/lxl-3086-classificationCanada.groovy

```diff
@@ -0,0 +1,61 @@
+PrintWriter scheduledForUpdating = getReportWriter("scheduled-updates")
+PrintWriter failedUpdating = getReportWriter("failed-updates")
+
+String where = "data#>'{@graph,2,marc:hasClassificationNumbersAssignedInCanada}' is not null"
+
+selectBySqlWhere(where) { data ->
+    def (record, instance, work) = data.graph
+
+    // force @graph,1,classification to be a list
+    if (instance["classification"] != null) {
+        if (!(instance["classification"] instanceof List))
+            instance["classification"] = [instance["classification"]]
+    } else
+        instance["classification"] = []
+
+    // Copy the work classifications
+    instance["classification"].addAll( getObjectList(work["marc:hasClassificationNumbersAssignedInCanada"]) )
+
+    // Ditch the work classifications
+    work.remove("hasClassificationNumbersAssignedInCanada")
+
+    // Clean up classifications
+    for (Object classification : instance["classification"]) {
+        if ( !(classification instanceof Map) )
+            continue
+
+        classification.remove("marc:existenceInLacCollection")
+        classification.remove("marc:sourceOfCallClassNumber")
+
+        if (classification["marc:itemNumber"] != null) {
+            classification["itemPortion"] = classification["marc:itemNumber"]
+            classification.remove("marc:itemNumber")
+        }
+
+        if (classification["marc:classificationNumber"] != null) {
+            classification["classificationPortion"] = classification["marc:classificationNumber"]
+            classification.remove("marc:classificationNumber")
+        }
+
+        if (classification["marc:typeCompletenessSourceOfClassCallNumber"] != null && (
+                classification["marc:typeCompletenessSourceOfClassCallNumber"] == "marc:LcBasedCallNumberAssignedByLac" ||
+                        classification["marc:typeCompletenessSourceOfClassCallNumber"] == "marc:CompleteLcClassNumberAssignedByLac" ||
+                        classification["marc:typeCompletenessSourceOfClassCallNumber"] == "marc:IncompleteLcClassNumberAssignedByLac" )) {
+            classification["source"] = classification["marc:typeCompletenessSourceOfClassCallNumber"]
+        }
+        classification.remove("marc:classificationNumber")
```

Comment:
Isn't it `marc:typeCompletenessSourceOfClassCallNumber` instead that you want to delete here?

Reply:
Good catch! ;) I fix!